ruma-api-macros/src/api/response.rs

```diff
@@ -290,52 +240,34 @@ impl ToTokens for Response {
             quote! { { #(#fields),* } }
         };
 
-        let (derive_deserialize, def) =
-            if let Some(body_field) = self.fields.iter().find(|f| f.is_newtype_body()) {
-                let field = Field { ident: None, colon_token: None, ..body_field.field().clone() };
+        let def = if let Some(body_field) = self.fields.iter().find(|f| f.is_newtype_body()) {
+            let field = Field { ident: None, colon_token: None, ..body_field.field().clone() };
 
-                // Though we don't track the difference between new type body and body
-                // for lifetimes, the outer check and the macro failing if it encounters
-                // an illegal combination of field attributes, is enough to guarantee
-                // `body_lifetimes` correctness.
-                let (derive_deserialize, lifetimes) = if self.has_body_lifetimes() {
-                    (TokenStream::new(), self.body_lifetimes())
-                } else {
-                    (quote!(::ruma_api::exports::serde::Deserialize), TokenStream::new())
-                };
-
-                (derive_deserialize, quote! { #lifetimes (#field); })
-            } else if self.has_body_fields() {
-                let fields = self.fields.iter().filter(|f| f.is_body());
-                let (derive_deserialize, lifetimes) = if self.has_body_lifetimes() {
-                    (TokenStream::new(), self.body_lifetimes())
-                } else {
-                    (quote!(::ruma_api::exports::serde::Deserialize), TokenStream::new())
-                };
+            quote! { (#field); }
+        } else if self.has_body_fields() {
+            let fields = self.fields.iter().filter(|f| f.is_body());
 
-                let fields = fields.map(ResponseField::field);
+            let fields = fields.map(ResponseField::field);
 
-                (derive_deserialize, quote!( #lifetimes { #(#fields),* }))
-            } else {
-                (TokenStream::new(), quote!({}))
-            };
+            quote!( { #(#fields),* } )
+        } else {
+            quote! { {} }
+        };
 
         let response_body_struct = quote! {
             /// Data in the response body.
             #[derive(
                 Debug,
                 ::ruma_api::Outgoing,
+                ::ruma_api::exports::serde::Deserialize,
                 ::ruma_api::exports::serde::Serialize,
-                #derive_deserialize
             )]
             struct ResponseBody #def
         };
 
-        let response_generics = self.combine_lifetimes();
         let response = quote! {
-            #[derive(Debug, Clone, ::ruma_api::Outgoing)]
-            #[incoming_no_deserialize]
-            pub struct Response #response_generics #response_def
+            #[derive(Debug, Clone, ::ruma_api::Outgoing, ::ruma_api::exports::serde::Deserialize)]
```

Comment:
`Request` and `Response` should never derive `Deserialize` or `Serialize`. Please re-add `#[incoming_no_deserialize]` and remove the `Deserialize` derive.

As a followup, we may want to remove the `Outgoing` derive and `Response: Outgoing` bound too. But probably in a separate PR.